File: library/src/main/java/me/shkschneider/skeleton/ui/LongPreference.java
package me.shkschneider.skeleton.ui;

import android.annotation.TargetApi;
import android.content.Context;
import android.preference.Preference;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import me.shkschneider.skeleton.helper.AndroidHelper;

public class LongPreference extends Preference {

    public LongPreference(final Context context) {
        super(context);
    }

    public LongPreference(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public LongPreference(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @SuppressWarnings("unused")
    @TargetApi(AndroidHelper.API_21)
    public LongPreference(final Context context, final AttributeSet attrs, final int defStyleAttr, final int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onBindView(final View view) {
        super.onBindView(view);
        ((TextView) view.findViewById(android.R.id.summary)).setMaxLines(5);
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/ui/LongPreference.kt
package me.shkschneider.skeleton.ui

import android.annotation.TargetApi
import android.content.Context
import android.preference.Preference
import android.util.AttributeSet
import android.view.View
import android.widget.TextView

import me.shkschneider.skeleton.helper.AndroidHelper

class LongPreference : Preference {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    @TargetApi(AndroidHelper.API_21)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    override fun onBindView(view: View) {
        super.onBindView(view)
        (view.findViewById<View>(android.R.id.summary) as TextView).maxLines = 5
    }

}
